src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs

```diff
@@ -1154,6 +1156,40 @@ private static void ReadMSBuildMetadata(JsonTextReader jsonReader, PackageSpec p
 
                         warningProperties = new WarningProperties(warnAsError, noWarn, allWarningsAsErrors, warningsNotAsErrors);
                         break;
+                    case "SdkAnalysisLevel":
+                        string skdAnalysisLevelString = jsonReader.ReadNextTokenAsString();
+
+                        if (!string.IsNullOrEmpty(skdAnalysisLevelString))
+                        {
+                            try
+                            {
+                                sdkAnalysisLevel = new NuGetVersion(skdAnalysisLevelString);
+                            }
+                            catch (ArgumentException ex)
+                            {
+                                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Strings.Invalid_AttributeValue, "SdkAnalysisLevel", skdAnalysisLevelString, "9.0.100"), ex);
```

Comment:
That's how I had it implemented at first, but after @zivkan 's comments I added the custom exceptions. I agree it could be excessive but in the case somehow a dgspec file is edited and it contains an invalid value for these attributes, we will get an informative error telling the user which property exactly is causing the failure. The version parser would produce a general error on the other hand which might not be as helpful